### PR TITLE
add alternative to rocm_smi_lib

### PR DESCRIPTION
**Pull request by edgargabriel**
## Details

Introduce an alternative mechanism to detect devices and links independent of rocm-smi. Usage of the rocm-smi library is still available as a fall-back solution and can be enforced by setting `RCCL_USE_ROCM_SMI_LIB=1` 

**Work item:** _"Internal", or link to GitHub issue (if applicable)._
LWPCOMMLIBS-260

**What were the changes?**  

1. parsing information available in the `/sys/class/kfd/kfd/topology/nodes/*/properties` and in `/sys/class/kfd/kfd/topology/nodes/*/io_links/*/properties`  
2. integration of the alt_rsmi functionality into rocm_smi_wrap.cc

**Why were the changes made?**  
fix an issue observed on some platforms

**How was the outcome achieved?**  
-

**Additional Documentation:**  
tested locally on MI100, MI250X, MI300X in SPX and MI300X in CPX mode

## Approval Checklist
___Do not approve until these items are satisfied.___
- [ ] Verify the CHANGELOG has been updated, if
  - there are any NCCL API version changes,
  - any changes impact library users, and/or
  - any changes impact any other ROCm library.

### CODE REVIEW

**Review by wenkaidu — APPROVED** [review]

File: src/include/alt_rsmi.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ *
+ * See LICENSE.txt for license information
+ ************************************************************************/
+
+#ifndef __ALT_RSMI_H__
+#define __ALT_RSMI_H__
+
+/*
+** This is a light-weight implementation of the RSMI functionality used in RCCL
+** The code is based on the actual rocm_smi_library code, but extract to contain only
```
Comment (gilbertlee-amd) [inline src/include/alt_rsmi.h]: extract-ed?

**Review by PedramAlizadeh — APPROVED** [review]

File: src/include/alt_rsmi.h (hunk 1 above)
Comment (edgargabriel) [reply src/include/alt_rsmi.h]: I will fix that, thank you!

**Comment by edgargabriel** [comment]
Hm, some CIs are failing with an error: 

`alt_rsmi.cc:233 NCCL WARN Error reading link properties files`

I cannot reproduce that locally, will have to debug this.

**Comment by edgargabriel** [comment]
the failed CI seems unrelated to this PR

```
Error: docker: Error response from daemon: error while creating mount source path '/var/jenkins_home': mkdir /var/jenkins_home: read-only file system.
```

**Pull request merged**